src/libs/actions/Session.js

```diff
@@ -208,19 +208,21 @@ function restartSignin() {
  *
  * @param {String} password
  * @param {String} validateCode
+ * @param {String} email
  */
-function setPassword(password, validateCode) {
+function setPassword(password, validateCode, email) {
     Onyx.merge(ONYXKEYS.ACCOUNT, {error: '', loading: true});
 
     API.SetPassword({
-        email: credentials.login,
+        email: email || credentials.login,
         password,
         validateCode,
     })
         .then((response) => {
             if (response.jsonCode === 200) {
-                const {authToken, email} = response;
-                createTemporaryLogin(authToken, email);
+                const {authToken} = response;
+                const login = response.email;
```

Comment:
NAB, we can just use these directly in `createTemporaryLogin()` instead of creating variables for them.